### PR TITLE
Remap individuals after union

### LINKED ISSUE #1225: union needs to remap individual parent IDs
To do this, we need to first build the map of which individual IDs are shared between `self` and `other`, by looping over the shared nodes, [here](https://github.com/tskit-dev/tskit/blob/20c423f055f03b84c98980e6e99febba9a6217d9/c/tskit/tables.c#L9887).  (And, we don't have tests with individual parents to test this.)

(See also #1223.)

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #1225.

### CODE REVIEW

File: c/tskit/tables.c [hunk 1]
```diff
@@ -10026,7 +10026,20 @@ tsk_table_collection_union(tsk_table_collection_t *self,
     memset(individual_map, 0xff, other->individuals.num_rows * sizeof(*individual_map));
     memset(population_map, 0xff, other->populations.num_rows * sizeof(*population_map));
     memset(site_map, 0xff, other->sites.num_rows * sizeof(*site_map));
-
+    for (k = 0; k < (tsk_id_t) other->individuals.num_rows; k++) {
+        individual_map[k] = TSK_NULL;
```
Comment (petrelharp) [inline c/tskit/tables.c:10030]: isn't this taken care of by the `memset` above?

Comment (benjeffery) [reply c/tskit/tables.c:10030]: Ah of course!

**Comment by benjeffery** [comment]
Just realised this isn't right as we should only remap the new individuals.

**Comment by jeromekelleher** [comment]
Maybe a few test cases in Python would help clarify what is required here?

**Comment by petrelharp** [comment]
Oh, right - I guess the individual parent remapping needs to happen during [add_and_remap_node](https://github.com/tskit-dev/tskit/pull/1238/files#diff-a7113e19292b94afe45f78ce230588105799e862a5f6f15d1af1ba81816f6e76R10048). Oh, wait, that won't work, because we don't know if we'll hit parent nodes before children.  Gah, this stuff is terrible.

Maybe the easiest thing is to separate out building the node table from building the individual table? First loop over nodes and build the individual map but don't actually add the rows to the individual table; then go through afterwards and add in the individuals?

**Comment by benjeffery** [comment]
I think I have it now - remap after all the `add_and_remap_node` calls, but only loop over the new individuals. Should have a PR with tests tonight.

**Comment by benjeffery** [comment]
Testing this is making me think `wf_sim` should always default to `record_individuals=True`. I'm going through the test failures from that now.

**Comment by benjeffery** [comment]
I've just pushed a commit that does this, and it required changing tests to check the metadata of individuals. Seems a good change.

**Review by jeromekelleher — APPROVED** [review]

**Comment by benjeffery** [comment]
@hyanwong getting an svg error here. Is it possible the svg isn't deterministic?

**Comment by hyanwong** [comment]
> @hyanwong getting an svg error here. Is it possible the svg isn't deterministic?

What's the error? I can't imagine how SVG creation would fail to be deterministic, although I guess it could depend on something unexpected such as e.g. the node order. I'll double-check that the test suite uses random seeds for all the generated tree.I had some SVG changes when switching to the new msprime version, so of which seemed like differences in floating point rounding (although I have no idea why that might happen in this case)

**Comment by benjeffery** [comment]
Ahhh, the nodes have individual classes! And as this PR adds individuals by default to wf sim the svg has changed. False alarm!

**Comment by hyanwong** [comment]
Phew. If the sim has changed, you can overwrite the old SVG file like this:

```
pytest tests/test_drawing.py::TestDrawSvg::test_known_svg_ts_multiroot --overwrite-expected-visualizations
```

**Comment by benjeffery** [comment]
Yeah, that flag is awesome!

**Pull request merged**